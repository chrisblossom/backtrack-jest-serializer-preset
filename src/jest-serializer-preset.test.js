import { transformConfig } from '@backtrack/core/dist/options-file/transform-config';

describe('serializer', () => {
    test('removes hash from files dest object', () => {
        const backtrackConfig = {
            packageJson: { outside: true },
            files: [
                { src: '__sandbox__/file1.js', dest: 'package.json' },
                { src: '__sandbox__/file2.js', dest: '.babelrc.js' },
            ],
        };

        const result = transformConfig(backtrackConfig, __dirname);

        expect(result).toMatchSnapshot();
    });
});
