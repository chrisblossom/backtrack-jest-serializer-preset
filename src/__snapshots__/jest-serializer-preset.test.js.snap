// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer removes hash from files dest object 1`] = `
Object {
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".babelrc.js": "<PROJECT_ROOT>/.babelrc.js",
        "package.json": "<PROJECT_ROOT>/package.json",
      },
      "allowChanges": Object {
        ".babelrc.js": false,
        "package.json": false,
      },
      "files": Array [
        "package.json",
        ".babelrc.js",
      ],
      "ignoreUpdates": Object {
        ".babelrc.js": false,
        "package.json": false,
      },
    },
    "makeDirs": Array [
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "src/__sandbox__/file1.js": "<PROJECT_ROOT>/src/__sandbox__/file1.js",
        "src/__sandbox__/file2.js": "<PROJECT_ROOT>/src/__sandbox__/file2.js",
      },
      "files": Array [
        "src/__sandbox__/file1.js",
        "src/__sandbox__/file2.js",
      ],
      "hash": Object {
        "src/__sandbox__/file1.js": "49598ea59816da0629609bb00775082fb21490af94fbcc8fbbbb120a6f5dc135",
        "src/__sandbox__/file2.js": "6cfc52d273559d0815ab3c56e1a1ad59b540948ddff9162991599c068d55e46a",
      },
    },
  },
  "packageJson": Array [
    Object {
      "outside": true,
    },
  ],
}
`;
